src/pending_connection/connect.rs

```diff
@@ -71,8 +71,8 @@ pub enum ConnectError {
 }
 
 pub struct Connect {
-    config: ConnectConfiguration,
-    state: ConnectState,
+    pub config: ConnectConfiguration,
+    pub state: ConnectState,
```

Comment:
I am leaving the fields on the *Configuration structs public. I believe I made these changes so that I could extract this code to the protocol module. It don't think it's essential, ultimately I think there is a better design that doesn't require access to the handshake state, perhaps as a return value.